cmd/acra-translator/server/server.go

```diff
@@ -193,129 +187,216 @@ func (server *ReaderServer) HandleConnectionString(parentContext context.Context
 	var outErr error
 	select {
 	case <-parentContext.Done():
-		log.WithError(parentContext.Err()).Debugln("Exit from handling connection string. Close all connections")
 		outErr = parentContext.Err()
 	case outErr = <-errCh:
 		log.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorTranslatorCantAcceptNewHTTPConnection).
 			Errorln("Error on accepting new connections")
-		server.Stop()
 	}
 	return outErr
 }
 
-// Constants show possible connection types.
-const (
-	ConnectionTypeKey  = "connection_type"
-	HTTPConnectionType = "http"
-	GRPCConnectionType = "grpc"
-)
+// Start setups gRPC handler or HTTP handler, poison records callbacks and starts listening to connections.
+func (server *ReaderServer) Start(parentContext context.Context) {
+	defer server.exitWithTimeout()
+
+	logger := logging.GetLoggerFromContext(parentContext)
+	poisonCallbacks := base.NewPoisonCallbackStorage()
+	server.detectPoisonRecords(poisonCallbacks)
+	errCh := make(chan error)
 
-type handlerName struct{}
+	decryptorData := &common.TranslatorData{Keystorage: server.keystorage, PoisonRecordCallbacks: poisonCallbacks, CheckPoisonRecords: server.config.DetectPoisonRecords()}
+	if server.config.IncomingConnectionHTTPString() != "" {
+		listener, err := network.Listen(server.config.IncomingConnectionHTTPString())
+		if err != nil {
+			log.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorTranslatorCantAcceptNewHTTPConnection).
+				Errorln("Can't create HTTP listener from specified connection string")
+			return
+		}
+		server.listenerHTTP = listener
+		server.startHTTP(parentContext, logger, decryptorData, errCh, listener)
+	}
 
-func withHandlerName(ctx context.Context, name string) context.Context {
-	return context.WithValue(ctx, handlerName{}, name)
-}
+	// provide way to register new services and custom server
+	if server.config.IncomingConnectionGRPCString() != "" {
+		listener, err := network.Listen(server.config.IncomingConnectionGRPCString())
+		if err != nil {
+			log.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorTranslatorCantAcceptNewHTTPConnection).
+				Errorln("Can't create HTTP listener from specified connection string")
```

Comment:
copy paste detected :)
```suggestion
				Errorln("Can't create gRPC listener from specified connection string")
```